source/server-discovery-and-monitoring/tests/unified/logging-standalone.json

```diff
@@ -0,0 +1,522 @@
+                    "topology": "debug"
+                  },
+                  "observeEvents": [
+                    "serverHeartbeatFailedEvent"
+                  ],
+                  "uriOptions": {
+                    "appname": "failingHeartbeatLoggingTest"
+                  }
+                }
+              }
+            ]
+          }
+        },
+        {
+          "name": "failPoint",
+          "object": "testRunner",
+          "arguments": {
+            "client": "setupClient",
+            "failPoint": {
+              "configureFailPoint": "failCommand",
+              "mode": "alwaysOn",
+              "data": {
+                "failCommands": [
+                  "hello",
+                  "isMaster"
+                ],
+                "appName": "failingHeartbeatLoggingTest",
+                "closeConnection": true
+              }
+            }
+          }
+        },
+        {
+          "name": "waitForEvent",
+          "object": "testRunner",
+          "arguments": {
+            "client": "client",
+            "event": {
+              "serverHeartbeatFailedEvent": {}
+            },
+            "count": 1
+          }
+        }
+      ],
+      "expectLogMessages": [
+        {
+          "client": "client",
+          "ignoreMessages": [
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Topology description changed"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Stopped topology monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Stopped server monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Topology description changed"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Starting server monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Starting topology monitoring"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Server heartbeat started"
+              }
+            },
+            {
+              "level": "debug",
+              "component": "topology",
+              "data": {
+                "message": "Server heartbeat succeeded"
+              }
+            }
+          ],
+          "messages": [
```

Comment:
The expected messages assertion here is failing in Rust because we see multiple heartbeat failed events I think this is reasonable/expected behavior: the failpoint is always on, so as the monitor continues to send `hello`s/listen for `hello` replies and emit heartbeat failed events when they fail. I can make the test work either by adding `ignoreExtraMessages` here or changing the fail point mode above to `{ "times": 2 }` (2 instead of 1 to account for the retry) rather than always on. Does either of those sound like fine change here, or am I missing something?

Reply:
Good catch, thanks. Added `ignoreExtraMessages`.